src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php

```diff
@@ -66,23 +67,55 @@ public static function uploadAudioFile($projectId, $mediaType, $tmpFilePath)
             "audio/x-mpg",
             "audio/x-mpegaudio",
             "audio/x-wav",
-            "audio/wav"
+            "audio/wav",
+            "audio/ogg",
+            "audio/webm",
+            "video/webm"
         );
         $allowedExtensions = array(
             ".mp3",
             ".mpa",
             ".mpg",
             ".m4a",
-            ".wav"
+            ".wav",
+            ".ogg",
+            ".webm"
         );
 
         $response = new UploadResponse();
+
         if (in_array(strtolower($fileType), $allowedTypes) && in_array(strtolower($fileExt), $allowedExtensions)) {
 
             // make the folders if they don't exist
             $project->createAssetsFolders();
             $folderPath = $project->getAudioFolderPath();
 
+            // convert audio file to mp3 or wav format if necessary //LHM
+            // FLEx only supports mp3 or wav format as of 2022-09
+
+            // First, find duration of audio file
+            // @return string
+            function getDuration($path) {
```

Comment:
I want to be sure all of these variables are either generated and trusted by our code or if there is user input being used, it's properly sanitized since this stuff is being released to the shell...

Reply:
Thanks. I took out the helper function and put those lines of code in the uploadAudioFile method itself. The only new variables introduced store the shell command and the resulting duration from an audio recording. User input is not used. Duration is calculated off of a variable (the filepath) that was already in the code.

Reply:
ok, thanks!